shared_model/utils/variant_deserializer.hpp

```diff
@@ -40,8 +48,9 @@ namespace shared_model {
          * @tparam Archive container type
          */
         template <class V, class T = typename V::types, class Archive>
-        static V invoke(Archive &&, int) {
+        NORETURN static V invoke(Archive &&, int) {
           BOOST_ASSERT_MSG(false, "Required type not found");
+          std::exit(1);
```

Comment:
Imo it's better to use `std::abort`, so 1) either all of the would be closed 2) it's possible to catch SIGABRT
Also there's a constant for that `EXIT_FAILURE`